File: 0868-binary-gap/0868-binary-gap.java
class Solution {
    public int binaryGap(int n) {
        String binary = Binary(n);
        
        int i = 0, j = 1, cnt = 0;
        while(j<binary.length()){
            if(binary.charAt(j)=='0'){
                j++;
            }
            else{
                cnt = (cnt>(j-i)) ? cnt : (j-i);
                i = j;
                j++;
            }
        }

        return cnt;
    }

    public String Binary(int n){
        String res = "";

        while(n>1){
            int rem = n%2;
            res = rem+res;
            n = n/2;
        }

        res = "1"+res;

        return res;
    }
}

File: 0868-binary-gap/0868-binary-gap.py
class Solution:
    def binaryGap(self, n: int) -> int:
        binary = self.Binary(n)
        
        i, j, cnt = 0, 1, 0
        while(j<len(binary)):
            if(binary[j]=="0"):
                j+=1
            else:
                cnt = max(cnt, (j-i))
                i = j
                j += 1
        
        return cnt

    def Binary(self, n):
        res = ""

        while(n>1):
            rem = n%2
            res = str(rem)+res
            n = n//2
        
        res = "1"+res
        
        return res